cache_ui/self_monitor.go

```diff
@@ -0,0 +1,236 @@
+package cache_ui
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/pelicanplatform/pelican/config"
+	"github.com/pelicanplatform/pelican/metrics"
+	"github.com/pelicanplatform/pelican/param"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	selfTestBody string = "This object was created by the Pelican self-test functionality"
+	selfTestDir  string = "/pelican/monitoring"
+)
+
+func InitSelfTestDir() error {
+	uid, err := config.GetDaemonUID()
+	if err != nil {
+		return err
+	}
+
+	gid, err := config.GetDaemonGID()
+	if err != nil {
+		return err
+	}
+
+	basePath := param.Cache_DataLocation.GetString()
+	pelicanMonPath := path.Join(basePath, "/pelican")
```

Comment:
Use `filepath` for files on disk, `path` for URLs:

```suggestion
	pelicanMonPath := filepath.Join(basePath, "/pelican")
```

On Unix, they are the same but this will reduce the work we would have to do for any future Windows port.